api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -113,6 +113,16 @@ default Integer sortOrderId() {
     return null;
   }
 
+  /** Returns the data sequence number of the snapshot in which the file should be applied. */
+  default Long dataSequenceNumber() {
+    return null;
+  }
+
+  /** Returns the file sequence number. */
+  default Long fileSequenceNumber() {
+    return null;
```

Comment:
Shall we also implement these methods in `IndexedXXX` in `V1Metadata` and `V2Metadata` and delegate to the wrapper, just like we do for `specId`?